### ISSUE TITLE
False positive from UselessParentheses sniff

### ISSUE THREAD

**Issue by janlanger** [slevomat/coding-standard#509]
Following code is reported as having useless parentheses

```php
$var = $foo && ($bar > 0 || $baz > 0) ? ... : ...;
```
Parentheses are not useless because `$foo && $bar > 0 || $baz > 0` has different semantics.
Only happens when ternary operator is used.

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/8559df049ebb9dfd356de767797f037136fc9d76
